src/network/serverpackethandler.cpp

```diff
@@ -2064,3 +2065,73 @@ void Server::handleCommand_SrpBytesM(NetworkPacket* pkt)
 
 	acceptAuth(pkt->getPeerId(), wantSudo);
 }
+
+/*
+ * Mod channels
+ */
+
+void Server::handleCommand_ModChannelJoin(NetworkPacket *pkt)
+{
+	std::string channel_name;
+	*pkt >> channel_name;
+
+	NetworkPacket resp_pkt(TOCLIENT_MODCHANNEL_SIGNAL, 1 + 2 + channel_name.size(),
+		pkt->getPeerId());
+
+	// Send signal to client to notify join succeed or not
+	if (m_modchannel_mgr->joinChannel(channel_name, pkt->getPeerId())) {
+		resp_pkt << (u8)MODCHANNEL_SIGNAL_JOIN_OK;
+		infostream << "Peer " << pkt->getPeerId() << " joined channel " << channel_name
+				<< std::endl;
+	} else {
+		resp_pkt << (u8)MODCHANNEL_SIGNAL_JOIN_FAILURE;
```

Comment:
oh ok, why not just ignore the message if the client is already joined?

Reply:
maybe i can add a ifndef NDEBUG here too ?